test/defn/test-unicode.kore

```diff
@@ -1,4 +1,4 @@
-// RUN: %interpreter
+// RUN: LC_ALL=en_US.UTF-8 %interpreter
```

Comment:
Why don't we just ensure that the YAML is always read/written in UTF-8 regardless of the system locale?

```
diff --git a/matching/src/main/scala/org/kframework/backend/llvm/matching/dt/DecisionTree.scala b/matching/src/main/scala/org/kframework/backend/llvm/matching/dt/DecisionTree.scala
index 817ea0c..e0d9f1c 100644
--- a/matching/src/main/scala/org/kframework/backend/llvm/matching/dt/DecisionTree.scala
+++ b/matching/src/main/scala/org/kframework/backend/llvm/matching/dt/DecisionTree.scala
@@ -2,22 +2,29 @@ package org.kframework.backend.llvm.matching.dt
 
 import org.kframework.backend.llvm.matching.Occurrence
 import org.kframework.backend.llvm.matching.pattern._
-import java.io.File
-import java.io.FileWriter
+
+import java.io.{File, FileOutputStream, OutputStreamWriter}
 import java.util
 import java.util.concurrent.ConcurrentHashMap
-
 import org.yaml.snakeyaml.Yaml
 
+import java.nio.charset.Charset
+
 sealed trait DecisionTree {
   def serializeToYaml(file: File): Unit = {
-    val writer = new FileWriter(file)
+    val writer = new OutputStreamWriter(
+      new FileOutputStream(file),
+      Charset.forName("UTF-8").newEncoder()
+    );
     new Yaml().dump(representation, writer)
     writer.close()
   }
 
   def serializeToYaml(file: File, residuals: Seq[(Pattern[String], Occurrence)]): Unit = {
-    val writer = new FileWriter(file)
+    val writer = new OutputStreamWriter(
+      new FileOutputStream(file),
+      Charset.forName("UTF-8").newEncoder()
+    );
     val residualRepr = new util.ArrayList[AnyRef]()
     for (entry <- residuals) {
       val pair = new util.ArrayList[AnyRef]()
```

```
diff --git a/lib/codegen/DecisionParser.cpp b/lib/codegen/DecisionParser.cpp
index 9e9d191..0d80fde 100644
--- a/lib/codegen/DecisionParser.cpp
+++ b/lib/codegen/DecisionParser.cpp
@@ -362,6 +362,7 @@ DecisionNode *parseYamlDecisionTreeFromString(
   yaml_parser_t parser;
   yaml_document_t doc;
   yaml_parser_initialize(&parser);
+  yaml_parser_set_encoding(&parser, YAML_UTF8_ENCODING);
   yaml_parser_set_input_string(
       &parser, (unsigned char *)yaml.c_str(), yaml.size());
   yaml_parser_load(&parser, &doc);
```